cadcutils/cadcutils/net/ws.py

```diff
@@ -455,84 +455,40 @@ def upload_file(self, url, src, md5_checksum=None, **kwargs):
            :throws: HttpExceptions
         """
         stat_info = os.stat(src)
-        with_transactions = False
+        if stat_info.st_size == 0:
+            raise ValueError('Cannot upload empty files')
         headers = kwargs.get('headers') or {}
         if not headers:
             kwargs['headers'] = headers
+        headers[HTTP_LENGTH] = str(stat_info.st_size)
         if md5_checksum:
             net.add_md5_header(headers=headers, md5_checksum=md5_checksum)
         else:
-            if stat_info.st_size >= MAX_MD5_COMPUTE_SIZE:
-                # pre-computing the MD5 is too costly.
-                # use PUT with transactions instead
-                headers[HTTP_TRANS] = 'true'
-                headers[HTTP_LENGTH] = str(stat_info.st_size)
-                with_transactions = True
-            else:
+            if stat_info.st_size <= MAX_MD5_COMPUTE_SIZE:
                 hash_md5 = hashlib.md5()
                 with open(src, "rb") as f:
                     for chunk in iter(lambda: f.read(4096), b""):
                         hash_md5.update(chunk)
-                md5 = hash_md5.hexdigest()
+                        md5 = hash_md5.hexdigest()
```

Comment:
Should this be calculated for every iteration?

Reply:
Involuntary change. Reverted